integration-tests/src/test/java/oracle/kubernetes/operator/ITSessionMigration.java

```diff
@@ -0,0 +1,345 @@
+    String sessCreateTime2 = getHttpResponseAttribute(result.stdout(), sessCreateTime);
+
+    // Verify that the same session info is used
+    Assume.assumeTrue("HTTP Session should NOT change!", sessCreateTime1.equals(sessCreateTime2));
+
+    // Verify that a new primary server is picked
+    Assume.assumeFalse(
+        "A new primary server should be picked!",
+        primaryServName1.trim().equals(primaryServName2.trim()));
+
+    // Restore test env
+    // Wait some time for ms pod to become ready
+    Thread.sleep(30 * 1000);
+    TestUtils.checkPodReady(domainUid + "-" + primaryServName1, domainNS);
+
+    logger.info(
+        "SUCCESS - " + testMethodName + ". ms <" + primaryServName2 + "> is new primary server.");
+  }
+
+  /**
+   * Verify that HTTP session state is migrated to the new primary server after the primary server
+   * shut down
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testHttpSessionMigr() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    Map<String, Object> domainMap = domain.getDomainMap();
+    String domainNS = domainMap.get("namespace").toString();
+    String domainUid = domain.getDomainUid();
+
+    int counterNum = 4;
+    String testAppPath = testAppName + "/CounterServlet";
+    String webServiceSetUrl = testAppPath + "?setCounter=" + counterNum;
+    String webServiceGetUrl = testAppPath + "?getCounter";
+
+    String primaryServ = "primary";
+    String count = "count";
+
+    // Send the first HTTP request and save HTTP header/sessionID info
+    ExecResult result = getHTTPResponse(webServiceSetUrl, " -D ");
+
+    // Get primary server name & count number bf primaryServName1 is stopped
+    String primaryServName1 = getHttpResponseAttribute(result.stdout(), primaryServ);
+    String countattribute1 = getHttpResponseAttribute(result.stdout(), count);
+
+    // Stop primary server
+    stopPrimaryServer(primaryServName1);
+
+    // Send the second HTTP request using HTTP header/sessionID info save before
+    result = getHTTPResponse(webServiceGetUrl, " -b ");
+
+    // Get primary server name & count number af primaryServName1 is stopped
+    String primaryServName2 = getHttpResponseAttribute(result.stdout(), primaryServ);
+    String countattribute2 = getHttpResponseAttribute(result.stdout(), count);
+
+    // Verify that the count number is from a new primary server
+    Assume.assumeFalse(
+        "A new primary server should be picked!",
+        primaryServName1.trim().equals(primaryServName2.trim()));
+
+    // Verify that HTTP session state is migrated by checking the count number same
+    // bf and af primaryServName1 stopped
+    Assume.assumeTrue(
+        "HTTP session state is NOT migrated!", countattribute1.equals(countattribute2));
+
+    // Restore test env
+    // Wait some time for ms pod to become ready
+    Thread.sleep(30 * 1000);
```

Comment:
sleep 30 secs is not needed here, TestUtils.checkPodReady checks in loop. This adds up test run time. @hzhao-github @anpanigr

Reply:
agreed. However, if there no any waiting time, the test will start before the pod become ready. Tried 10 seconds butnot working.

Reply:
Can you explicitly wait for the pod to become ready rather than just a blind 30 seconds?  We have code that does this.

Reply:
Hi Ryan, entire block of that part looks like:

    // Restore test env
    // Wait some time for ms pod to become ready
    Thread.sleep(30 * 1000);
    TestUtils.checkPodReady(domainUid + "-" + primaryServName1, domainNS);

TestUtils.checkPodReady waits for 50 seconds for a pod gets ready. Since this test delete the pod, most likely, 50 econds is not long enough for the pod become up and running online again.

How about this, Vanaja if I make this changes:

In BaseTest.java, add a new set method to change

private static int maxIterationsPod = 50;

to a longer time, like maxIterationsPod = 90;

Reply:
why only 50 seconds? TestUtils.checkPodReady waits upto 250 seconds for the pod to become ready in loop. See below props defined in Operator.props file. 
```
maxIterationsPod=25
waitTimePod=10
```

Reply:
I was wrong. Yes, we wait up to 250 seconds :)